### PR TITLE
Expose and use HarfBuzz's mmap for font loading

### LINKED ISSUE #52: Use harfbuzz's mmap
uharfbuzz currently suggest reading fonts like, 
```
with open(sys.argv[1], 'rb') as fontfile:
    fontdata = fontfile.read()

text = sys.argv[2]

face = hb.Face(fontdata)
font = hb.Font(face)
upem = face.upem
```

but that's sub-optimal, one can enable HAVE_MMAP https://github.com/harfbuzz/harfbuzz/blob/master/src/hb-blob.cc#L478-L485 for non-Windows platform (Windows is already fine) and use `hb_blob_create_from_file` instead, I wonder however is uharfbuzz used in places that lots of fonts are loaded thus the it worth the font loading optimization.

### PR DESCRIPTION
**Pull request by ebraminio**
Fixes #52

### CODE REVIEW

**Comment by ebraminio** [comment]
This one needs a little discussion for API naming and so, some feedback would be nice of course.

**Comment by justvanrossum** [comment]
> This one needs a little discussion for API naming and so, some feedback would be nice of course.

I think `from_file_path` is fine. The PR looks good to me, but I'd prefer for someone with more knowledge have a look.

File: setup.py [hunk 1]
```diff
@@ -22,6 +22,9 @@
 extra_compile_args = []
 if platform.system() != 'Windows':
```
Comment (ebraminio) [inline setup.py:23]: Just to note Windows is already fine and able to find things needed for mmap.

**Comment by ebraminio** [comment]
> I think `from_file_path` is fine. The PR looks good to me, but I'd prefer for someone with more knowledge have a look.

Sure :)

**Comment by khaledhosny** [comment]
I’d rather we review the API and make it closer to the C API, e.g. make blob creation a separate step, create face only from blobs and deprecate passing data directly to it.

**Comment by anthrotype** [comment]
> I’d rather we review the API and make it closer to the C API, e.g. make blob creation a separate step, create face only from blobs and deprecate passing data directly to it.

I agree. Can you file a separate issue for this?

**Comment by khaledhosny** [comment]
> > I’d rather we review the API and make it closer to the C API, e.g. make blob creation a separate step, create face only from blobs and deprecate passing data directly to it.
> 
> I agree. Can you file a separate issue for this?

I’ll try. But since I’m slow to respond, please don’t let this block any ongoing development.

**Comment by khaledhosny** [comment]
Sorry, closed by mistake.

**Comment by khaledhosny** [comment]
Superseded by https://github.com/harfbuzz/uharfbuzz/pull/99

**Pull request closed without merging**